src/extractor/guidance/turn_handler.cpp

```diff
@@ -632,12 +644,11 @@ boost::optional<TurnHandler::Fork> TurnHandler::findFork(const EdgeID via_edge,
     {
         // makes sure that the fork is isolated from other neighbouring streets on the left and
         // right side
-        const auto next =
-            (fork->left + 1) == intersection.end() ? intersection.begin() : (fork->left + 1);
+        const auto next = fork->end == intersection.end() ? intersection.begin() : (fork->end);
         const bool separated_at_left_side =
-            angularDeviation(fork->left->angle, next->angle) >= GROUP_ANGLE;
+            angularDeviation(fork->getLeft().angle, next->angle) >= GROUP_ANGLE;
         const bool separated_at_right_side =
-            angularDeviation(fork->right->angle, (fork->right - 1)->angle) >= GROUP_ANGLE;
+            angularDeviation(fork->getRight().angle, (fork->begin - 1)->angle) >= GROUP_ANGLE;
```

Comment:
are we sure that `begin` is safe to subtract from here? Should be due to the u-turn, right?